### ISSUE TITLE
[data grid] Pressing Tab should not leave the cell while editing a date / dateTime field

### ISSUE THREAD

**Issue by flaviendelangle** [mui/mui-x#2930]
### Current behavior 😯

Follow up on #2854 

On a `date` or `datetime-local` input, we use the tab key to switch among the date portions (from the year to the month for instance)

The current editing behavior is to move to the next cell when pressing Tab, which makes the input uncomfortable to edit.

### Expected behavior 🤔

The browser input should behave like here https://developer.mozilla.org/fr/docs/Web/HTML/Element/Input/datetime-local#valeur

To approach it, I see several solutions

### Solution 1

Only switch to the next cell when we are already editing the last portion of the date.
But the browser API do not seem to tell us which portion is currently highlighted, and even if we had them, it may be buggy depending on the date format.

### Solution 2

Never switch to the next cell with "Tab" on the `GridEditDateCell`

### Solution 3

Leave as is

____

I love none of those solutions, I you have a better idea I am open to suggestions

### Steps to reproduce 🕹

Steps:

1. Go to https://deploy-preview-2918--material-ui-x.netlify.app/components/data-grid/editing/#cell-editing
2. Start editing a cell in the `Last login` column
3. Press `Tab`

**Comment by m4theushw** [comment]
> On a date or datetime-local input, we use the tab key to switch among the date portions (from the year to the month for instance)

This is not the default behavior of `<input type="date" />`. The arrow keys are used to navigate between the date components. I don't think we should try to override how <kbd>Tab</kbd> behaves. Pressing it should move the focus to the next focusable element and this is what the grid is doing.

**Comment by flaviendelangle** [comment]
https://user-images.githubusercontent.com/3309670/142369923-aac606e3-61a2-4348-a50c-e43226ce8216.mp4

When I press `Tab` on a `datetime-local`, I am switching between the date portions.
And it does not work on the grid because we leave the cell.

**Comment by m4theushw** [comment]
On Edge, <kbd>Tab</kbd> moves the focus to the next focusable element. However, in Chrome and Firefox, they allow to navigate between the date components instead. Each browser implements the date input in a different way. There's no specification saying how keyboard navigation should work in these inputs. In this case, I think they should follow https://www.w3.org/TR/wai-aria-practices/#kbd_general_within. Weirdly, tabbing inside a `<input type="number" />` in Firefox doesn't move the focus between the step up/down buttons but it goes to the next focusable element. 🤷‍♂️

The native date input has caused many problems in the past. I don't think it's worth to try to make the <kbd>Tab</kbd> work here. We should be using the date picker once it's available. If the user is implementing a custom component, he can stop the propagation and handle this key.

**Comment by alexfauquette** [comment]
I agree that if the behavior is not consistent between browser, we should keep simple and respect what the [documentation says](https://deploy-preview-2918--material-ui-x.netlify.app/components/data-grid/editing/#stop-editing) 
> -  A <kbd>Escape</kbd> keydown. It will also roll back changes done in the value of the cell.

@flaviendelangle should we close this issue?
